### DISCUSSION TITLE
Is 'getCollectionNode()' a public stable interface for Collections?

### DISCUSSION THREAD

**Discussion by nicholasrice** [adobe/react-spectrum#6341] [General]
Hi! First, thanks for all your hard work on react-aria, we're really enjoying using it :)

I'm working on a UI library that is using react-aria and react-stately hooks for most of the business logic. We're looking to create some proprietary Collection Item types. Internally, I see prolific use of [getCollectionNode](https://github.com/adobe/react-spectrum/search?q=getCollectionNode) which seems to be the main interface for interacting a Collection, however I don't see `getCollectionNode()` exposed publicly, nor is it referenced in documentation as far as I can tell. Is `getCollectionNode()` a stable interface that should be used for building proprietary Item types, or is that a private implementation detail of react-aria that shouldn't be used by library consumers?

**Comment by snowystinger** [comment]
In general, no. We don't publicly document it and don't consider it a public API. That said, it hasn't changed in a long time.

We do hope to eventually have a public Collection implementation, but it's likely to be the one from RAC. Though you may find you don't need to do anything so complex if you are using RAC.

https://react-spectrum.adobe.com/react-aria/Select.html#reusable-wrappers

And there's some work on exposing custom renderers here [Add support for custom collection renderers (e.g. virtualization)](https://github.com/adobe/react-spectrum/pull/5912#top) which may further mitigate your issues

**Reply by nicholasrice** [reply]
Okay thank you for the resources!
